core/controllers/question_test.py

```diff
@@ -111,3 +111,350 @@ def test_get(self):
             '%s/batch' % feconf.QUESTION_DATA_URL, self.payload,
             expect_errors=True)
         self.assertEqual(response.status_int, 404)
+
+
+class QuestionsHandlerTest(test_utils.GenericTestBase):
+    """Test the question handler."""
+
+    def setUp(self):
+        super(QuestionsHandlerTest, self).setUp()
+
+        self.collection_id = 'coll_0'
+        self.exp_id = 'exp_1'
+        self.signup(self.NEW_USER_EMAIL, self.NEW_USER_USERNAME)
+        self.new_user_id = self.get_user_id_from_email(self.NEW_USER_EMAIL)
+
+        # Create a new collection and exploration.
+        self.save_new_valid_collection(
+            self.collection_id, self.new_user_id, exploration_id=self.exp_id)
+
+        # Add a skill.
+        collection_services.update_collection(
+            self.new_user_id, self.collection_id, [{
+                'cmd': collection_domain.CMD_ADD_COLLECTION_SKILL,
+                'name': 'test'
+            }], 'Add a new skill')
+        collection = collection_services.get_collection_by_id(
+            self.collection_id)
+        self.skill_id = collection.get_skill_id_from_skill_name('test')
+        collection_node = collection.get_node(self.exp_id)
+        collection_node.update_acquired_skill_ids([self.skill_id])
+
+        # Update the acquired skill IDs for the exploration.
+        collection_services.update_collection(
+            self.new_user_id, self.collection_id, [{
+                'cmd': collection_domain.CMD_EDIT_COLLECTION_NODE_PROPERTY,
+                'property_name': (
+                    collection_domain.COLLECTION_NODE_PROPERTY_ACQUIRED_SKILL_IDS), # pylint: disable=line-too-long
+                'exploration_id': self.exp_id,
+                'new_value': [self.skill_id]
+            }], 'Update skill')
+
+        self.payload = {}
```

Comment:
Defining a self.payload in setUp() doesn't really add anything. Just have a 'payload' var in each test, or construct the dict manually inline. The less stuff you put on "self", the better -- it just adds to the amount of context that needs to be carried around.

Ditto for other test classes.

Reply:
Updated.